libs/domain/picking/offline/services/sw-push-initializer.service.ts

```diff
@@ -0,0 +1,29 @@
+import { AppInitializer } from '@spryker-oryx/core';
+import { inject } from '@spryker-oryx/di';
+import { SyncSchedulerService } from '@spryker-oryx/offline';
+import {
+  PickingSyncAction,
+  PushSyncPayload,
+} from '@spryker-oryx/picking/offline';
+import { firstValueFrom, Observable } from 'rxjs';
+
+declare let self: ServiceWorkerGlobalScope;
+
+export class SwPushInitializerService implements AppInitializer {
+  constructor(private syncSchedulerService = inject(SyncSchedulerService)) {}
+
+  initialize(): void | Observable<void> | Promise<void> {
+    self.addEventListener('push', (event: PushEvent) => {
+      const payload: PushSyncPayload = event.data?.json();
```

Comment:
Here is optional chaining in event data. Can be a case when there is not data? If so - should we schedule such sync?

Reply:
```suggestion
      const payload: PushSyncPayload = event.data.json();
```